src/components/Editor.js

```diff
@@ -40,6 +42,8 @@ const SAVED_TAG_FILTERS_DATALIST_ID = 'saved_tag_filters_datalist';
 class Editor extends React.Component {
 
   state = {
+    editorKey: '',
+    setEditorValue: '',
```

Comment:
Previously, I calculated the editor value directly in the render function, which re-rendered whenever the current file id state changed, but I don't think that allows using await/.then to return the JSX, so it needs to load the new source content value in componentDidUpdate (which is called before the render function), and then update these two state fields before rendering (calling setState in componentDidUpdate should be fine). 

The editor component only actually rerenders when its key prop is changed, so we need to set this key via this editorKey state. This is because during rendering, the value is only set as the defaultvalue rather than the value itself, because the value prop has some weird behavior/side effects.